src/components/carousel/_index.scss

```diff
@@ -153,4 +152,32 @@
 		}
 		@include scss.component-properties("carousel-indicator-thumbnail");
 	}
+
+	&:fullscreen {
+		@include scss.component-properties("carousel-fullscreen");
+
+		// setting fullscreen status for counter label
+		// overrides counter label styles
+		.c-carousel {
+			&__image-counter-label {
+				@include scss.component-properties("carousel-fullscreen-image-counter");
+			}
+
+			&__button {
+				> * {
+					@include scss.component-properties("carousel-button-full-screen-content");
+				}
+
+				&:hover {
+					@include scss.component-properties("carousel-button-full-screen-hover");
+
+					> * {
+						@include scss.component-properties("carousel-button-full-screen-hover-content");
+					}
+				}
+
+				@include scss.component-properties("carousel-button-full-screen");
+			}
```

Comment:
Like I mentioned in the blocks ticket, did you try using `"components": { "button": {} }` to apply the styling for these?

Reply:
Comment attended on refactoring.